moveit_core/planning_scene/src/planning_scene.cpp

```diff
@@ -514,10 +536,17 @@ void PlanningScene::checkCollisionUnpadded(const collision_detection::CollisionR
   // check collision with the world using the unpadded version
   getCollisionEnvUnpadded()->checkRobotCollision(req, res, robot_state, acm);
 
-  // do self-collision checking with the unpadded version of the robot
+  // do self-collision checking with the unpadded version of the robot - unless otherwise specified
   if (!res.collision || (req.contacts && res.contacts.size() < req.max_contacts))
   {
-    getCollisionEnvUnpadded()->checkSelfCollision(req, res, robot_state, acm);
+    if (use_padding_self_collisions_)
+    {
+      getCollisionEnv()->checkSelfCollision(req, res, robot_state, acm);
+    }
+    else
+    {
+      getCollisionEnvUnpadded()->checkSelfCollision(req, res, robot_state, acm);
+    }
```

Comment:
Revert this. `checkCollisionUnpadded` is used to check the remaining path during trajectory execution, as used below:
https://github.com/sbgisen/moveit/blob/1fbb2c4c555fe59cb316fea184760468aa702d66/moveit_ros/planning/plan_execution/src/plan_execution.cpp#L297-L315

I don`t know why the author/maintainer does this, but I assume the padded collision check is unnecessary if the planning has already performed it. If you want to have MoveIt check padded collision even during execution, modify the code above instead.

Reply:
I see. That's right.